swift/ql/test/query-tests/Security/CWE-611/testAEXMLDocumentXXE.swift

```diff
@@ -0,0 +1,148 @@
+// --- stubs ---
+
+class Data {
+    init<S>(_ elements: S) {}
+}
+
+struct URL {
+	init?(string: String) {}
+}
+
+extension String {
+    struct Encoding: Hashable {
+        let rawValue: UInt
+        static let utf8 = String.Encoding(rawValue: 1)
+    }
+
+	init(contentsOf: URL) {
+        let data = ""
+        self.init(data)
+    }
+}
+
+class AEXMLElement {}
+
+struct AEXMLOptions {
+    var parserSettings = ParserSettings()
+
+    struct ParserSettings {
+        public var shouldResolveExternalEntities = false
+    }
+}
+
+class AEXMLDocument {
+    init(root: AEXMLElement? = nil, options: AEXMLOptions) {}
+    init(xml: Data, options: AEXMLOptions = AEXMLOptions()) {}
+    init(xml: String, encoding: String.Encoding, options: AEXMLOptions) {}
```

Comment:
The `AEXML` library itself has some defaults you don't specify here.  Your test cases below obviously don't use them.  Can you imagine it matters?

Reply:
The empty `AEXMLOptions` and `String.Encoding` = `utf8` would hardly affect the results. At most, we could verify that we correctly have FNs when the default `options:` argument is used, but the test would look very similar to the ones explicitly setting an empty `AEXMLOptions` (i.e. the `*safeImplicit` test cases). I'm open to adding those additional cases nevertheless, if you think they are worth it.

Reply:
I'm happy how it is.  But I have wondered when implementing tests myself whether I should put effort into modelling and using default values accurately to how the libraries do it.  I think you're right there isn't much difference between using a default and using a non-default value that's the same as it anyway.  The query certainly doesn't care if a default is there / used.